nca/Parsers/IstioGatewayPolicyGenerator.py

```diff
@@ -0,0 +1,293 @@
+            return []
+
+        result = []
+        used_gateways = set()
+        for vs in self.vs_parser.virtual_services.values():
+            global_vs_gtw_to_hosts = self.pick_vs_gateways_by_hosts(vs, vs.gateway_names)
+            used_gateways.update(set(global_vs_gtw_to_hosts.keys()))
+            only_local_traffic = False
+            vs_policies = self.create_route_policies(vs, vs.http_routes, global_vs_gtw_to_hosts, used_gateways,
+                                                     only_local_traffic)
+            vs_policies.extend(self.create_route_policies(vs, vs.tls_routes, global_vs_gtw_to_hosts, used_gateways,
+                                                          only_local_traffic))
+            vs_policies.extend(self.create_route_policies(vs, vs.tcp_routes, global_vs_gtw_to_hosts, used_gateways,
+                                                          only_local_traffic))
+            if vs_policies:
+                result.extend(vs_policies)
+            else:
+                if only_local_traffic:
+                    self.vs_parser.warning(f'The virtual service {vs.full_name()} defines '
+                                           f'only local (mesh-to-mesh) traffic that is ignored', vs)
+                else:
+                    self.vs_parser.warning(f'The virtual service {vs.full_name()} does not affect traffic '
+                                           f'and is ignored', vs)
+
+        unused_gateways = self.gtw_parser.get_all_gateways() - used_gateways
+        if unused_gateways:
+            self.vs_parser.warning(f'the following gateways have no virtual services attached: '
+                                   f'{",".join([gtw.full_name() for gtw in unused_gateways])}')
+        if not result:
+            self.vs_parser.warning('no valid VirtualServices found. Ignoring istio ingress traffic')
+        else:
+            result[0].findings = self.gtw_parser.warning_msgs + self.vs_parser.warning_msgs
+        return result
+
+    def pick_vs_gateways_by_hosts(self, vs, gateway_names):
+        """
+        Given a list of gateway names, retrieves the gateways that match the virtual service by hosts
+        or gives a warning for non-existing or non-matching ones
+        :param VirtualService vs: the virtual service that references the gateways in the list
+        :param list[str] gateway_names: the given gateway names list
+        :return: dict (Gateway, MinDFA): the map from gateways to the matching hosts.
+        """
+
+        gtw_to_hosts = {}
+        vs_all_hosts_dfa = reduce(MinDFA.__or__, vs.hosts_dfa)
+        for gtw_name in gateway_names:
+            if gtw_name == 'mesh':
+                continue
+            gtw = self.gtw_parser.get_gateway(gtw_name)
+            if gtw:
+                matching_hosts = gtw.all_hosts_dfa & vs_all_hosts_dfa
+                if matching_hosts:
+                    gtw_to_hosts[gtw] = matching_hosts
```

Comment:
add comment about how vs is matched to gw (not only by vs specifying the gw, also by hosts matching)